src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/InMemoryPartitionedAppTokenCacheAccessor.cs

```diff
@@ -0,0 +1,200 @@
+            // if a conflict occurs, pick the latest value
+            AccessTokenCacheDictionary
+                .GetOrAdd(partitionKey, new ConcurrentDictionary<string, MsalAccessTokenCacheItem>())[itemKey] = item;
+        }
+
+        /// <summary>
+        /// This method is not supported for the app token cache because
+        /// there are no refresh tokens, ID tokens, or accounts in a client credential flow
+        /// </summary>
+        public void SaveRefreshToken(MsalRefreshTokenCacheItem item)
+        {
+            throw new NotSupportedException();
+        }
+
+        /// <summary>
+        /// This method is not supported for the app token cache because
+        /// there are no refresh tokens, ID tokens, or accounts in a client credential flow
+        /// </summary>
+        public void SaveIdToken(MsalIdTokenCacheItem item)
+        {
+            throw new NotSupportedException();
+        }
+
+        /// <summary>
+        /// This method is not supported for the app token cache because
+        /// there are no refresh tokens, ID tokens, or accounts in a client credential flow
+        /// </summary>
+        public void SaveAccount(MsalAccountCacheItem item)
+        {
+            throw new NotSupportedException();
+        }
+
+        public void SaveAppMetadata(MsalAppMetadataCacheItem item)
+        {
+            string key = item.GetKey().ToString();
+            AppMetadataDictionary[key] = item;
+        }
+        #endregion
+
+        #region Get
+        /// <summary>
+        /// This method is not supported for the app token cache because
+        /// there are no refresh tokens, ID tokens, or accounts in a client credential flow
+        /// </summary>
+        public MsalIdTokenCacheItem GetIdToken(MsalAccessTokenCacheItem accessTokenCacheItem)
+        {
+            throw new NotSupportedException();
+        }
+
+        /// <summary>
+        /// This method is not supported for the app token cache because
+        /// there are no refresh tokens, ID tokens, or accounts in a client credential flow
+        /// </summary>
+        public MsalAccountCacheItem GetAccount(MsalAccountCacheKey accountKey)
+        {
+            throw new NotSupportedException();
+        }
+
+        public MsalAppMetadataCacheItem GetAppMetadata(MsalAppMetadataCacheKey appMetadataKey)
+        {
+            AppMetadataDictionary.TryGetValue(appMetadataKey.ToString(), out MsalAppMetadataCacheItem cacheItem);
+            return cacheItem;
+        }
+        #endregion
+
+        #region Delete
+        public void DeleteAccessToken(MsalAccessTokenCacheItem item)
+        {
+            var partitionKey = CacheKeyFactory.GetClientCredentialKey(item.ClientId, item.TenantId);
+
+            AccessTokenCacheDictionary.TryGetValue(partitionKey, out var partition);
+            if (partition == null || !partition.TryRemove(item.GetKey().ToString(), out _))
+            {
+                _logger.InfoPii(
```

Comment:
why is this log message Pii?

Reply:
GetKey has home account id, which is Pii

Reply:
do we need the cache key for logging? I don't find it particularly helpful. then we don't have to have pii logs anywhere here.